### ISSUE TITLE
Add support for wsrep_max_ws_rows limit

### ISSUE THREAD

**Issue by sjaakola** [mariadb-corporation/mysql-wsrep#257]
Currently write set limit happens by provider options variable: repl.max_ws_size
However, there is a need to limit the max write set size by number of rows modified in the transaction.

There are two related old launchpad bugs:
https://bugs.launchpad.net/codership-mysql/+bug/1373909
https://bugs.launchpad.net/codership-mysql/+bug/1087614

**Comment by ayurchen** [comment]
Limiting writeset by the number of rows may be not so useful as it may seem. What is used in certification is KEYS, and there will be more than one key per row. So it is not easy to predict the number of keys in a writeset from inside mysqld. Sometimes more than 2 or three.
Plus I think we have a solution for that in streaming replication.

In any case, it is a provider limit, so it should be controlled in provider, not in application.

**Comment by kbauskar** [comment]
I agree with Alex. I think it is not a good idea to limit the write-set base on number-of-rows. It would be difficult to predict how many number of rows will cross the other threshold enforced by max_ws_size.

User may set some value for max_ws_rows but still the load may fail if max_ws_size threshold is hit. This will be more annoying.

I guess users has got use to max_ws_size so may be we should continue to support only that.
(Of-course I assume in due-course this limit can be relaxed too with large-trx support).

**Comment by PrzemekMalkowski** [comment]
I can see the support was already added according to: http://galeracluster.com/2016/08/announcing-galera-cluster-5-5-50-and-5-6-31-with-galera-3-17/
